app/src/main/java/com/owncloud/android/utils/ShortcutUtils.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Nextcloud Android client application
+ *
+ * @author Felix Nüsse
+ *
+ * Copyright (C) 2022 Felix Nüsse
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.utils
+
+import com.owncloud.android.datamodel.OCFile
+import android.content.pm.ShortcutManager
+import android.content.Intent
+import com.owncloud.android.ui.activity.FileDisplayActivity
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.datamodel.ThumbnailsCacheManager
+import android.content.pm.ShortcutInfo
+import android.app.PendingIntent
+import android.app.PendingIntent.FLAG_IMMUTABLE
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Icon
+import android.os.Build
+import kotlin.math.roundToInt
+
+class ShortcutUtil(val mContext: Context) {
```

Comment:
```suggestion
class ShortcutUtil @Inject constructor (val mContext: Context) {
```
This will allow you to then inject the `ShortcutUtil` in `OCFileListFragment` and wherever else is needed.

Additionally the file should be name to `ShortcutUtil.kt` or the class to `ShortcutUtils`, to have them both match.